### ISSUE TITLE
updateServiceWorker does nothing

### ISSUE THREAD

**Issue by Mehdi-Hp** [antfu/vite-plugin-pwa#57]
I'm using `injectManifest` mode and this is my config
```
VitePWA({
    strategies: 'injectManifest',
    manifest,
    srcDir: './src/service-worker',
    filename: 'index.js',
    outDir: './dist',
    injectRegister: 'auto'
}),
```

And later on to implement the ReloadPrompt component,

```
<button v-if="newVersionAvailable" @click="updateServiceWorker">
    Update to the latest version
</button>
const {
    needRefresh
    updateServiceWorker
} = useRegisterSW();
```

But `updateServiceWorker` practically does nothing.   
It just returns a promise that resolves to nothing.  
I looked into the source code and saw `if (!auto) {` in `updateServiceWorker` function.  
Then saw this `const autoUpdateMode = '__SW_AUTO_UPDATE__'` which `updateServiceWorker` is depend on.  Then, found out that `__SW_AUTO_UPDATE__` is getting replaced by `${options.registerType === 'autoUpdate'}`. But I don't want an autoUpdate when I'm using a reload prompt. I lost.

**Comment by userquin** [comment]
@Mehdi-Hp take a look at https://github.com/antfu/vite-plugin-pwa/tree/master/examples/vue-basic-inject-manifest, see its `vite.config.ts` configuration file

**Comment by Mehdi-Hp** [comment]
@userquin Oh thanks. I can update the docs if you think it's necessary.

**Comment by userquin** [comment]
@Mehdi-Hp you can make a PR, I'll try to review it

We need to include a more detailed documentation, it is complex to understand how the service worker works.
Perhaps we could include a first version and debug it, keep in mind that there are a lot of configurations, from the configuration of the plugin itself through the workbox configuration.
If we add to the above, injecting a custom service worker (this issue), then anyone who does not have the knowledge of the internal functioning of the service worker will have many problems. 

@antfu what do you think? Right now I'm very busy, maybe next week...

**Comment by userquin** [comment]
@Mehdi-Hp docs included on PR #75
